Automattic/jetpack
# PHPCS: Update directory structure, and modernize class usage

#### Changes proposed in this Pull Request:

In starting to add another sniff, I noticed several inconsistencies in
the configuration of the current MasterUserConstant sniff.

* We seem to want our phpcs standard to be named "Jetpack", but phpcs was
  calling it "PHPCSSniffs".
* The MasterUserConstant sniff produces warnings with ID
  "..Jetpack_Sniffs_MasterUserConstant.ShouldNotBeUsed", rather than
  "Jetpack.Something.MasterUserConstant.ShouldNotBeUsed" like all other
  sniffs.
* MasterUserConstantSniff.php is using old phpcs 2.x aliases for the
  phpcs classes, which don't get loaded if you use phpcs --standard to
  run only Jetpack's sniffs.

This patch fixes the above with the following changes:

* As phpcs seems to want the standard to be in an eponymous directory,
  move bin/PHPCSSniffs to bin/PHPCSStandards/Jetpack.
* Rename Jetpack_Sniffs_MasterUserConstantSniff to
  Jetpack\Sniffs\Constants\MasterUserConstantSniff so the ID is
  determined correctly.
* Update MasterUserConstantSniff.php to use the phpcs 3.x class names.
* Move MasterUserConstantSniff.php to a PSR-4 style directory structure,
  since even the WordPress phpcs standards do that.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Check this out, and run `composer install` to refresh the phpcs configuration.
* Verify that "Jetpack" rather than "PHPCSSniffs" shows up from `composer php:lint -- -i`
* Add a use of the `JETPACK_MASTER_USER` constant to a file, and run `composer php:lint -- --standard=Jetpack file.php`. Verify that it works, and reports the warning with code "Jetpack.Constants.MasterUserConstant.ShouldNotBeUsed".

#### Proposed changelog entry for your changes:

* As this is a developer-only change, no changelog entry should be needed.
